robot_smach_states/src/robot_smach_states/human_interaction/find_people_in_room.py

```diff
@@ -0,0 +1,532 @@
+import smach
+
+# TU/e Robotics
+import robot_smach_states as states
+import robot_smach_states.util.designators as ds
+from robot_skills.util import kdl_conversions
+from robot_skills.util.entity import Entity
+
+__all__ = ['FindPeople', 'FindFirstPerson', 'SetPoseFirstFoundPersonToEntity', 'FindPeopleInRoom']
+
+
+class FindPeople(smach.State):
+    """
+    Smach state to find a person. The robot looks around and tries to find
+    people in its view.
+
+    >>> # The requried imports on robot console
+    >>> import robot_smach_states as states
+    >>> import robot_smach_states.util.designators as ds
+    >>> from robot_skills.util.entity import Entity
+    >>>
+    >>> # Designator to store the result
+    >>> des = ds.VariableDesignator(resolve_type=[Entity])
+    >>>
+    >>> sm = state.FindPeople(robot=robot, properties={'id': 'NAME'}, found_people_designator=des.writeable)
+    >>> sm.execute()
+    >>> des.resolve()
+    Entity(id=something, )
+
+    """
+
+    def __init__(self, robot, properties=None, query_entity_designator=None,
+                 found_people_designator=None, look_distance=10.0, speak=False,
+                 strict=True, nearest=False, attempts=1, search_timeout=60,
+                 look_range=(-np.pi/2, np.pi/2), look_steps=8):
+        """ Initialization method
+        :param robot: robot api object
+        :param properties: (dict) (default: None) keyvalue pair of the properties a person must
+            possess. None as a value for a property would search for all possible
+            values of the property.
+        :param query_entity_designator: (default: None) An entity designator to match all found
+            people to
+        :param found_people_designator: (default: None) A designator to write the search result to.
+            The designator always has a list of found people written to it.
+        :param look_distance: (float) (default: 10.0) The distance (radius) which the robot must look at
+        :param speak: (bool) (default: False) If True, the robot will speak while trying to find
+            a named person
+        :param strict: (bool) (default: True)  Only used if properties is not None AND the {key:value} pair of a
+        property has non None values.
+            If set to True then only people with all specified
+            properties are returned, else all people with at least one true property.
+            Example:
+                properties = {'tags': ['LWave', 'RWave', 'LHolding', 'RHolding']}
+                strict = True
+                    This will return a list of people who have the tags:
+                        'LWave' AND 'RWave' AND 'LHolding' AND 'RHolding'
+
+                strict = False
+                    This will return a list of people who have the tags:
+                        'LWave' OR 'RWave' OR 'LHolding' OR 'RHolding'
+
+        :param nearest: (bool) (default: False) If True, selects the people nearest to the robot who match the
```

Comment:
I think the function activated with the `nearest` parameter is not something that should be implemented on this level. This function should just return a list of found people. I would leave sorting those people to a higher statemachine.